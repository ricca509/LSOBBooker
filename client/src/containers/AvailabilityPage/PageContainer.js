import React, { Component } from 'react';
import { observer, inject } from 'mobx-react';
import { fetchData } from '../../actions';
import Page from './Page';

export class PageContainer extends Component {
  componentDidMount() {
    const { store } = this.props;

    if (store.isReadyToFetch) {
      fetchData();
    }
  }

  render() {
    const { store } = this.props;

    return (
      <Page
        availability={store.availability}
        fetching={store.fetching}
        fetched={store.fetched}
        fetchError={store.fetchError}
        hasAvailability={store.hasAvailability}
        locationId={store.selectedLocationId}
        serviceId={store.selectedEventId}
      />
    );
  }
}

export default inject('store')(observer(PageContainer));
