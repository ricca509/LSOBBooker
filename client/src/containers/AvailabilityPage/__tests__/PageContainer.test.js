import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { PageContainer } from '../PageContainer';

describe('PageContainer', () => {
  it('renders the Page with the right data', () => {
    const wrapper = shallow(<PageContainer store={{
      availability: [{
        date: 'date',
        times: [1, 2]
      }],
      hasAvailability: true,
      location: 4
    }} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
