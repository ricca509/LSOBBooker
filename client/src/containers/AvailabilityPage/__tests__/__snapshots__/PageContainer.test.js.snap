exports[`PageContainer renders the Page with the right data 1`] = `
<Page
  availability={
    Array [
      Object {
        "date": "date",
        "times": Array [
          1,
          2,
        ],
      },
    ]
  }
  hasAvailability={true}
  locationId={4}
  serviceId={68} />
`;
